packages/redux/src/users/closets/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,281 @@
+const randomAction = { type: 'this_is_a_random_action' };
+
+describe('closets reducer', () => {
+  beforeEach(() => {
+    initialState = fromReducer.INITIAL_STATE;
+  });
+
+  describe('reset handling', () => {
+    it('RESET_USER_CLOSETS should return the initial state', () => {
+      expect(
+        reducer(mockState.closets, {
+          type: actionTypes.RESET_USER_CLOSETS,
+        }),
+      ).toEqual(initialState);
+    });
+  });
+
+  describe('error() reducer', () => {
+    it('should return the initial state', () => {
+      const state = reducer(undefined, randomAction).error;
+
+      expect(state).toBe(initialState.error);
+      expect(state).toBeNull();
+    });
+
+    it.each([actionTypes.FETCH_USER_CLOSETS_REQUEST])(
+      'should handle %s action type',
+      actionType => {
+        const state = {
+          ...initialState,
+          error: toBlackoutError(new Error('error')),
+        };
+
+        expect(
+          reducer(state, {
+            type: actionType,
+          }).error,
+        ).toBe(initialState.error);
+      },
+    );
+
+    it.each([actionTypes.FETCH_USER_CLOSETS_FAILURE])(
+      'should handle %s action type',
+      actionType => {
+        const state = {
+          ...initialState,
+          error: null,
+        };
+        const error = 'foo';
+
+        expect(
+          reducer(state, {
+            payload: { error },
+            type: actionType,
+          }).error,
+        ).toBe(error);
+      },
+    );
+
+    it('should handle other actions by returning the previous state', () => {
+      const state = {
+        ...initialState,
+        error: toBlackoutError(new Error('error')),
+      };
+
+      expect(reducer(state, randomAction).error).toBe(state.error);
+    });
+  });
+
+  describe('isLoading() reducer', () => {
+    it('should return the initial state', () => {
+      const state = reducer(undefined, randomAction).isLoading;
+
+      expect(state).toBe(initialState.isLoading);
+      expect(state).toBe(false);
+    });
+
+    it.each([actionTypes.FETCH_USER_CLOSETS_REQUEST])(
+      'should handle %s action type',
+      actionType => {
+        const state = {
+          ...initialState,
+          isLoading: false,
+        };
+
+        expect(
+          reducer(state, {
+            type: actionType,
+          }).isLoading,
+        ).toBe(true);
+      },
+    );
+
+    it.each([actionTypes.FETCH_USER_CLOSETS_SUCCESS])(
+      'should handle %s action type',
+      actionType => {
+        const state = {
+          ...initialState,
+          isLoading: true,
+        };
+
+        expect(
+          reducer(state, {
+            payload: { result: '' },
+            type: actionType,
+          }).isLoading,
+        ).toBe(initialState.isLoading);
+      },
+    );
+
+    it.each([actionTypes.FETCH_USER_CLOSETS_FAILURE])(
+      'should handle %s action type',
+      actionType => {
+        expect(
+          reducer(undefined, {
```

Comment:
Use another value for state instead of undefined please to make the test more reliable.

Reply:
done